### PR TITLE
Add in tags support

### PR DESCRIPTION
**Pull request by iheanyi**
This is the first part of adding support for tags to `godotto`. Let me know if I have the right idea with regards to testing, wasn't sure if I was missing anything from it.

### CODE REVIEW

File: pkg/droplets/droplets_test.go [hunk 1]
```diff
@@ -132,7 +132,7 @@ var want = {
   snapshot_ids: [ 42 ],
   status: "loling",
   networks: {"v4":{}, "v6":{}},
-  tags: ["derp"],
```
Comment (aybabtme) [inline pkg/droplets/droplets_test.go]: why are you hating on the derps :(

File: pkg/droplets/droplets_test.go [hunk 2]
```diff
@@ -208,6 +208,7 @@ var d = pkg.create({
 	private_networking: true,
 	user_data:    "lolll",
 	monitoring:   true,
+	tags:					["test"],
```
Comment (aybabtme) [inline pkg/droplets/droplets_test.go]: them tabs! not to start a war but the rest was space so can you make this spaces too :)

File: pkg/droplets/droplets_test.go (hunk 1 above)
Comment (iheanyi) [inline pkg/droplets/droplets_test.go]: :trollface: Trololol.

File: pkg/extra/do/cloud/client.go [hunk 3]
```diff
@@ -97,3 +101,5 @@ func (svc *client) Regions() regions.Client         { return svc.regions }
 func (svc *client) Sizes() sizes.Client             { return svc.sizes }
 func (svc *client) FloatingIPs() floatingips.Client { return svc.floatingips }
 func (svc *client) Volumes() volumes.Client         { return svc.volumes }
+
```
Comment (aybabtme) [inline pkg/extra/do/cloud/client.go]: kill the empty space

File: pkg/droplets/droplets_test.go (hunk 2 above)
Comment (iheanyi) [inline pkg/droplets/droplets_test.go]: Yeah, good catch. Didn't see how it was formatted differently.

File: pkg/tags/tags.go [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+	tag := ottoutil.String(vm, arg)
+	err := svc.svc.Delete(svc.ctx, tag)
+	if err != nil {
+		ottoutil.Throw(vm, err.Error())
+	}
+
+	return q
+}
+
+func (svc *tagSvc) tag_resources(all otto.FunctionCall) otto.Value {
```
Comment (aybabtme) [inline pkg/tags/tags.go]: `s/tag_resources/tagResources/g`

File: pkg/tags/tags.go [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+
+	err = svc.svc.TagResources(svc.ctx, name, req.Resources)
+	if err != nil {
+		ottoutil.Throw(vm, err.Error())
+	}
+
+	return q
+}
+
+func (svc *tagSvc) untag_resources(all otto.FunctionCall) otto.Value {
```
Comment (aybabtme) [inline pkg/tags/tags.go]: same

File: pkg/tags/tags_test.go [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+	vmtest.Run(t, cloud, `
+		var pkg = cloud.tags;
+		pkg.delete("test");
+	`)
+}
+
+func TestTagTagResources(t *testing.T) {
+	cloud := mockcloud.Client(nil)
+	cloud.MockTags.TagFn = func(_ context.Context, name string, res []godo.Resource) error {
+		// Won't return an error
```
Comment (aybabtme) [inline pkg/tags/tags_test.go]: I think you'll want to assert that `name == "tests` and `res[0] == {...}`

File: pkg/tags/tags_test.go [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+				}	
+			]
+		});
+	`)
+}
+
+func TestTagUntagResources(t *testing.T) {
+	cloud := mockcloud.Client(nil)
+	cloud.MockTags.UntagFn = func(_ context.Context, name string, res []godo.Resource) error {
+		// Won't return an error
```
Comment (aybabtme) [inline pkg/tags/tags_test.go]: same here

**Comment by aybabtme** [comment]
Pretty good work here, just make sure to fill up the tag/untag tests and see if it works in the REPL too!

**Comment by iheanyi** [comment]
@aybabtme – It's ready for one more lookover whenever you have a chance. :)

**Comment by aybabtme** [comment]
Awesome thanks!

**Pull request merged**